server/accounts.go

```diff
@@ -831,13 +872,34 @@ func (s *Server) updateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 	s.Debugf("Updating account claims: %s", a.Name)
 	a.checkExpiration(ac.Claims())
 
+	a.mu.Lock()
 	// Clone to update, only select certain fields.
-	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits}
+	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits, signingKeys: a.signingKeys}
 
 	// Reset exports and imports here.
 	a.exports = exportMap{}
 	a.imports = importMap{}
 
+	// update account signing keys
+	a.signingKeys = nil
+	signersChanged := false
+	if len(ac.SigningKeys) > 0 {
+		// insure sorted and copy
+		sort.Strings(ac.SigningKeys)
+		a.signingKeys = append(a.signingKeys, ac.SigningKeys...)
+	}
+	if len(a.signingKeys) != len(old.signingKeys) {
+		signersChanged = true
+	} else {
+		for i, v := range old.signingKeys {
```

Comment:
Again, if arrays are supposed to be same size and order is supposed to be the same, I would use a `for i:=0; i<len(a.signingKeys);i++{` and compare directly `a.signingKeys[i] != old.signingKeys[i]` to avoid a copy.

Reply:
fixed